### ISSUE TITLE
ci: Improve Workflow for Final Release Builds w/ GoReleaser

### ISSUE THREAD

**Issue by CharlieMc0** [zeta-chain/node#1174]
We need an improved release workflow.

This release should target a tag, branch, or commit and have the option to include manually generated release notes. 

This task covers the Manual Approval > Final Release Testing > Publish Release portion of the flowchart 

![Image](https://github.com/zeta-chain/node/assets/31941002/c107b785-3cb1-49c2-9c14-a3b097a4d436)

AC
- Create a release is triggered by some manual action
- An approval process must pass before a release is published 
- Testing must be completed before a release is published 
- The tests themselves are out of scope for this task since all the tests currently lives at the PR level
- Binaries should be cross compiled and available for at least Linux and Mac in x86 and ARM.

**Comment by CharlieMc0** [comment]
This has been started using Goreleaser here - https://github.com/zeta-chain/node/tree/ci-goreleaser

**Comment by CharlieMc0** [comment]
GoReleaser has been added here https://github.com/zeta-chain/node/pull/1218

**Comment by lumtis** [comment]
Closing as it looks completed.
